### ISSUE TITLE
Crash during searching

### ISSUE THREAD

**Issue by Suncatcher** [TeamNewPipe/NewPipe#3106]
![photo_2020-02-17_09-54-58](https://user-images.githubusercontent.com/6388034/74633963-90264880-516b-11ea-8718-b2a8ae80d471.jpg)

Steps to reproduce:

1. Put search term in Cyrillic "настоящее время" (maybe not only this), and hit enter.

It doesn't happen always, just sometimes, I cannot figure any rule. I observe this bug throughout the last three app versions, I believe.

P.S. The screenshot is from the version 0.18.2 but I experienced it in 0.18.3 as well.

**Comment by cool-student** [comment]
this happened to me as well recently but as stated above it doesn't happen every time. for me it happened when I was scrolling down, possibly something didn't load quickly enough ?

## Exception
* __User Action:__ searched
* __Request:__ eminem
* __Content Language:__ GB
* __Service:__ YouTube
* __Version:__ v0.18.3
* __OS:__ Linux Android 10 - 29


<details><summary><b>Crash log</b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'org.jsoup.select.Elements org.jsoup.nodes.Element.children()' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.collectItems(YoutubeSearchExtractor.java:114)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.getPage(YoutubeSearchExtractor.java:92)
	at org.schabi.newpipe.extractor.search.SearchInfo.getMoreItems(SearchInfo.java:62)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getMoreSearchItems$1(ExtractorHelper.java:90)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$DBE2dF7XqVvkl-TcN5nthmVnvzQ.call(Unknown Source:10)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</p></details>
<hr>

**Comment by Stypox** [comment]
Could you get more information about the webpage with the apk provided in #1816?

**Comment by Suncatcher** [comment]
@Stypox  this app-debug.apk is from the 5th of October. Is it still actual? Won't it introduce more bugs?

**Comment by Stypox** [comment]
Mmmh maybe it crashes. I will try to provide a new apk when I have time. Please note that the apk's purpose is just that of crashing with more info, not actually fixing anything

**Comment by cool-student** [comment]
unfortunately this only happened to me once and I'm having a hard time reproducing it :/

**Comment by B0pol** [comment]
Reproduced it. It didn't amke the app crash though, I had no results and error at the bottom

## Exception
* __User Action:__ searched
* __Request:__ pewdiepie break
* __Content Language:__ FR
* __GMT Time:__ 2020-02-22 18:28
* __Package:__ org.schabi.newpipe.debug
* __Version:__ 0.18.3
* __Service:__ YouTube
* __Operating System:__ Linux Android 8.1.0 - 27


<details><summary><b>Crash log</b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'org.jsoup.select.Elements org.jsoup.nodes.Element.children()' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.collectItems(YoutubeSearchExtractor.java:114)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.getInitialPage(YoutubeSearchExtractor.java:79)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getItemsPageOrLogError(ExtractorHelper.java:20)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:50)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:30)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$searchFor$0(ExtractorHelper.java:77)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$BBduYDeZ_vXMQYaemaggmTPtqvA.call(Unknown Source:8)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)

```
</p></details>
<hr>

**Comment by mehedihasanziku** [comment]
Yes, true, it's happened with data network connection & when low internet connectivity. Though i can't reproduce that issue, i will open issue with log or update here when i will face it again.

**Comment by Suncatcher** [comment]
The same happens on 0.18.4. The problem persists

![photo_2020-02-25_16-08-10](https://user-images.githubusercontent.com/6388034/75254524-27814080-57e9-11ea-98cb-7620e8f125a8.jpg)

## Exception
* __User Action:__ searched
* __Request:__ настоящее время
* __Content Language:__ GB
* __Service:__ YouTube
* __Version:__ 0.18.4
* __OS:__ Linux xiaomi/onc/onc:9/PKQ1.181021.001/V11.0.2.0.PFLMIXM:user/release-keys 9 - 28


<details><summary><b>Crash log</b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'org.jsoup.select.Elements org.jsoup.nodes.Element.children()' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.collectItems(YoutubeSearchExtractor.java:114)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.getInitialPage(YoutubeSearchExtractor.java:79)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getItemsPageOrLogError(ExtractorHelper.java:19)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:50)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:30)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$searchFor$0(ExtractorHelper.java:77)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$BBduYDeZ_vXMQYaemaggmTPtqvA.call(Unknown Source:8)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)

```
</p></details>
<hr>

**Comment by wb9688** [comment]
@Suncatcher: Does this issue also happen with https://send.firefox.com/download/5b21cf1361768659/#NUmIIvWTZ1Lwr2D3CK6L7w? That's a build of NewPipe that parses YouTube's new website. That build isn't ready yet, so there are other issues with it, but if this issue doesn't happen there, this issue could be closed.

**Comment by Suncatcher** [comment]
@wb9688 Installation fails on Android 9

**Comment by B0pol** [comment]
Well just update to v0.18.5, it uses the new website

**Comment by Suncatcher** [comment]
Still had no update notification prompt. 
BTW, isn't it correct that my version show Content Lang: **GB**, even when I search content on Russian and my phone is on English (US) language, not GB?
Phone language shouldn't influence content, no?

**Comment by B0pol** [comment]
From [v0.18.4 blog post](https://newpipe.schabi.org/blog/release/pinned/newpipe-0.18.4-released/)

> @B0pol third contribution immensely improves the app for non-English speaker: The country NewPipe sets you in is now set to the one of your system language - No more UK videos in the Trending section if you are actually a proud Russian. This only applies to new installations, don’t worry if you enjoy your English feed, друг.

For content language: we put the same for the same reason. It should use your language instead of English, but it doesn't work currently, that's why you can see sometimes English titles or descriptions even though they aren't English videos (because it's translated).

**Comment by Suncatcher** [comment]
> but it doesn't work currently

and when will it work?

**Comment by Suncatcher** [comment]
On 0.18.6 version it still exists

![photo_2020-03-12_21-14-35](https://user-images.githubusercontent.com/6388034/76559079-2289e580-64a7-11ea-9c15-aace33135d06.jpg)

**Comment by Suncatcher** [comment]
Surprisingly today NewPipe opened live show, though usually it crashes even on videos added 40 minutes ago.

**Comment by Suncatcher** [comment]
I have this bug again on latest version

{
   "user_action": "requested stream",
   "request": "[https:\/\/www.youtube.com\/watch?v=1LvO7s4BjF8]https:\/\/www.youtube.com\/watch?v=1LvO7s4BjF8",
   "content_language": "GB",
   "service": "YouTube",
   "package": "org.schabi.newpipe",
   "version": "0.19.2",
   "os": "Linux Android 9 - 28",
   "time": "2020-04-16 19:13",
   "exceptions": [
      "java.lang.NullPointerException: Attempt to invoke virtual method
'com.grack.nanojson.JsonObject
com.grack.nanojson.JsonObject.getObject(java.lang.String)' on a null object
reference\n\tat
org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.getErrorMessage(YoutubeStreamExtractor.java:580)\n\tat
org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:80)\n\tat
org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:61)\n\tat
org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:120)\n\tat
org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(Unknown
Source:4)\n\tat
io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)\n\tat
io.reactivex.Single.subscribe(Single.java:3438)\n\tat
io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)\n\tat
io.reactivex.Single.subscribe(Single.java:3438)\n\tat
io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)\n\tat
io.reactivex.Maybe.subscribe(Maybe.java:4154)\n\tat
io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)\n\tat
io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)\n\tat
io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)\n\tat
io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)\n\tat
io.reactivex.Flowable.subscribe(Flowable.java:14479)\n\tat
io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)\n\tat
io.reactivex.Maybe.subscribe(Maybe.java:4154)\n\tat
io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)\n\tat
io.reactivex.Single.subscribe(Single.java:3438)\n\tat
io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)\n\tat
io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)\n\tat
io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)\n\tat
io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)\n\tat
java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)\n\tat
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n\tat
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\n\tat
java.lang.Thread.run(Thread.java:764)\n"
   ],
   "user_comment": ""
}

![photo_2020-04-18_09-48-49](https://user-images.githubusercontent.com/6388034/79630420-d2fba100-8159-11ea-83d5-17b7b9ec171f.jpg)

**Comment by Stypox** [comment]
Please use the [Incredible Error Report to Markdown Converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) before sending bug reports @Suncatcher

**Comment by Suncatcher** [comment]
On 0.19.3 the same

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=pjN_-nzzoiU
* __Content Language:__ GB
* __Service:__ YouTube
* __Version:__ 0.19.3
* __OS:__ Linux Android 9 - 28


<details><summary><b>Crash log</b></summary><p>

```
org.schabi.newpipe.extractor.stream.StreamInfo$StreamExtractException: Could not get any stream. See error variable to get further details.
	at org.schabi.newpipe.extractor.stream.StreamInfo.extractStreams(StreamInfo.java:192)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:70)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:62)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14479)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)

```
</p></details>
<hr>

**Comment by B0pol** [comment]
No it's not the same issue. As you can see in the crash log the error was NullPointerException, and in the last crash log you've sent, it's StreamExctractException
This bug is known: https://github.com/TeamNewPipe/NewPipe/issues/3576

**Comment by Suncatcher** [comment]
Sorry for confusing, posted in correspondent thread

**Comment by opusforlife2** [comment]
@Suncatcher Still there as of 0.19.8?

**Comment by Suncatcher** [comment]
Haven't seen it for a while

**Comment by opusforlife2** [comment]
Alright, thanks. Comment if you see it again.

**Comment by Suncatcher** [comment]
It happens again

## Exception
* __User Action:__ searched
* __Request:__ нихуя не понял но очень интересно умер
* __Content Language:__ en
* __Service:__ YouTube
* __Version:__ 0.20.0
* __OS:__ Linux Android 9 - 28


<details><summary><b>Crash log</b></summary><p>

```
org.schabi.newpipe.extractor.exceptions.ParsingException: Could not get ytInitialData
	at org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getInitialData(YoutubeParsingHelper.java:203)
	at org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.extractClientVersionAndKey(YoutubeParsingHelper.java:222)
	at org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getKey(YoutubeParsingHelper.java:298)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.getNewNextPageFrom(YoutubeSearchExtractor.java:233)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor.getInitialPage(YoutubeSearchExtractor.java:127)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getItemsPageOrLogError(ExtractorHelper.java:19)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:55)
	at org.schabi.newpipe.extractor.search.SearchInfo.getInfo(SearchInfo.java:30)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$searchFor$0(ExtractorHelper.java:81)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$BBduYDeZ_vXMQYaemaggmTPtqvA.call(Unknown Source:8)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: org.schabi.newpipe.extractor.utils.Parser$RegexException: failed to find pattern "window\["ytInitialData"\]\s*=\s*(\{.*?\});
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:72)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:61)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup1(Parser.java:52)
	at org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getInitialData(YoutubeParsingHelper.java:200)
	... 20 more

```
</p></details>
<hr>

**Comment by opusforlife2** [comment]
@Suncatcher Fixed in 0.20.1.
